rct229/rulesets/ashrae9012019/section5/section5rule22.py

```diff
@@ -99,10 +100,11 @@ def get_calc_vals(self, context, data=None):
                     return {
                         "has_shading_overhang": subsurface_b["has_shading_overhang"],
                         "has_shading_sidefins": subsurface_b["has_shading_sidefins"],
+                        "depth_of_overhang": subsurface_b["depth_of_overhang"],
                     }
 
                 def rule_check(self, context, calc_vals=None, data=None):
-                    return not (
-                        calc_vals["has_shading_overhang"]
-                        or calc_vals["has_shading_sidefins"]
-                    )
+                    return (
+                        calc_vals["has_shading_overhang"] is False
+                        or calc_vals["depth_of_overhang"] == ZERO.LENGTH
```

Comment:
The logic is quite twisted here. So in RDS it says ` Not(subsurface_b.has_shading_overhang OR subsurface_b.depth_of_overhang !=0)`
Apply De Morgan's Law:
`Not subsurface_b.has_shading_overhang AND Not subsurface_b.depth_of_overhang != 0`
So in this context. it is:
`calc_vals["has_shading_overhang"] is False and calc_vals["depth_of_overhang"] = ZERO.LENGTH.`

Reply:
Weili, I saw this in the RDS as well and it seemed like it was written incorrectly because if a window does not have a shading overhang then the overhang depth will not exist. 

I believe the intent was to test if the window has an overhang specified but the depth is 0, then it can be considered as flush with the wall.

Reply:
> h the pseudo-python logic.
> 
> I believe the intent was to test if the window has an overhang specified but the depth is 0, then it can be considered as flush with the wall. If you agree then I will submit a PR to update that line in the RDS.

Interesting, I never would consider a flush is a overhang but it is probably hard to tell from simulation software? In that case, please update the RDS to reflect the actual intent. Thanks for the clarification.

Reply:
@claperle will you please confirm if you agree with my interpretation of this rule, "it seemed like it was a mistake in the way that the pseudo-python was written because the logic written in English does not match the pseudo-python logic"

I am suggesting changing the python logic to:
(has_shading_overhang_b is False or depth_of_overhang_b == ZERO.LENGTH) and has_shading_sidefins_b is False

while the RDS has written: if subsurface is flush with the exterior wall, and no shading projections are modeled or projection depth is 0. Then described in pseudo-python: NOT (subsurface_b.has_shading_overhang OR subsurface_b.depth_of_overhang != 0)) AND ( NOT subsurface_b.has_shading_sidefins )

Reply:
@JacksonJ-KC I agree.